test/compilable/test22585.c

```diff
@@ -0,0 +1,4 @@
+// https://issues.dlang.org/show_bug.cgi?id=22585
+
+extern const unsigned char array[];
+const unsigned char array[4] = { 0, 1, 2, 3 };
```

Comment:
To be sure, this test should have the directives:
```
// REQUIRED_ARGS: -main
// LINK:
```

Reply:
I just moved it to `runnable`